### PR TITLE
#6932: Test yolov4 Downsample1 ttnn sub-module

### PR DESCRIPTION
**Pull request by saichandax**
This is a draft PR to discuss on the sub-module test of Downsample1 of YoloV4.

### CODE REVIEW

**Comment by saichandax** [comment]
@dvartaniansTT , The PR is updated.
Now, we have PCC ~0.996.
Please review.
Thank you.

File: models/experimental/functional_yolov4/reference/downsample1.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        x4_m = self.relu(x4_b)
+
+        x5 = self.c5(x4_m)
+        x5_b = self.b5(x5)
+        x5_m = self.relu(x5_b)
+
+        x6 = self.c6(x5_m)
+        x6_b = self.b6(x6)
+        x6_m = self.relu(x6_b)
+        x6_m = x6_m + x4_m
```
Comment (dvartaniansTT) [inline models/experimental/functional_yolov4/reference/downsample1.py:62]: @saichanda should we add x5_m and x3_m instead? Could you please take the model graph diagram and label it with the corresponding input and outputs, for instance the input to the very first conv labeled with "input" -> from the code. the output of it is x1 and so on. I have attached the graph for downsample1 here. we should do this for all the modules to ensure correct implementation. 

Screenshot 2024-04-09 at 6 28 11 PM

**Review by dvartaniansTT — CHANGES_REQUESTED** [review]
I've left an important comment which applies to all modules we are working on please check!

File: models/experimental/functional_yolov4/reference/downsample1.py (hunk 1 above)
Comment (keerthana-r-mcw) [reply models/experimental/functional_yolov4/reference/downsample1.py:62]: image

Comment (keerthana-r-mcw) [reply models/experimental/functional_yolov4/reference/downsample1.py:62]: downsample1_ref
downsample1_ttnn

**Review by dvartaniansTT — APPROVED** [review]

File: tests/ttnn/integration_tests/yolov4/test_ttnn_downsample1.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+                conv8_weight, conv8_bias, ttnn_module_args.c8, return_parallel_config=True
+            )
+
+        return parameters
+
+    return custom_preprocessor
+
+
+@skip_for_wormhole_b0()
+def test_downsample1(reset_seeds, device):
```
Comment (dvartaniansTT) [inline tests/ttnn/integration_tests/yolov4/test_ttnn_downsample1.py:165]: `def test_downsample1(device, reset_seeds):`
 and remove `device_id = 0`

**Review by dvartaniansTT — CHANGES_REQUESTED** [review]

**Review by dvartaniansTT — CHANGES_REQUESTED** [review]
please also follow these for testing: 

cd $TT_METAL_HOME
scripts/build_scripts/build_with_profiler_opt.sh
./tt_metal/tools/profiler/profile_this.py -n DS1 -c "pytest tests/ttnn/integration_tests/yolov4/test_ttnn_downsample1.py "

**Comment by keerthana-r-mcw** [comment]
Profiler of downsample1:
[ops_perf_results_DownSample1_2024_04_17_10_11_41.csv](https://github.com/tenstorrent/tt-metal/files/15009399/ops_perf_results_DownSample1_2024_04_17_10_11_41.csv)

File: tests/ttnn/integration_tests/yolov4/test_ttnn_downsample1.py (hunk 2 above)
Comment (saichandax) [reply tests/ttnn/integration_tests/yolov4/test_ttnn_downsample1.py:165]: This is still not removed @ankitmcw please update the commit. Changes are not reflected I suppose.

**Review by dvartaniansTT — APPROVED** [review]

**Review by arakhmati — APPROVED** [review]

**Comment by saichandax** [comment]
Nightly fast dispatch: [passed](https://github.com/tenstorrent/tt-metal/actions/runs/8822728124)
All post commit: [passed](https://github.com/tenstorrent/tt-metal/actions/runs/8822730218)

**Comment by saichandax** [comment]
nightly dispatch: [failed](https://github.com/tenstorrent/tt-metal/actions/runs/9007927404)
post commit: [passed](https://github.com/tenstorrent/tt-metal/actions/runs/9007930314)

**Comment by saichandax** [comment]
nightly fast dispatch: [passed](https://github.com/tenstorrent/tt-metal/actions/runs/9024450268)
post commit : [passed](https://github.com/tenstorrent/tt-metal/actions/runs/9024452110)

**Pull request closed without merging**